File: backend/models/time_entry.go
package models

import "time"

// TimeEntry represents a time tracking session for an activity
// EndTime is NULL when timer is still running
// PlannedDuration is NULL for free-running timers
// PausedAt is NULL when timer is not paused
type TimeEntry struct {
	ID              uint       `gorm:"primaryKey" json:"id"`
	UserID          uint       `gorm:"not null;index" json:"user_id"`
	ActivityID      uint       `gorm:"not null;index" json:"activity_id"`
	Activity        Activity   `gorm:"foreignKey:ActivityID" json:"activity,omitempty"`
	StartTime       time.Time  `gorm:"not null;index" json:"start_time"`
	EndTime         *time.Time `gorm:"index" json:"end_time,omitempty"`
	PlannedDuration *int       `gorm:"type:integer" json:"planned_duration,omitempty"` // Duration in seconds, NULL = free-running
	PausedAt        *time.Time `gorm:"index" json:"paused_at,omitempty"`               // When timer was paused, NULL = not paused
	PausedDuration  int        `gorm:"default:0" json:"paused_duration"`               // Cumulative seconds paused
	Notes           *string    `gorm:"type:text" json:"notes,omitempty"`
	CreatedAt       time.Time  `json:"created_at"`
}

File: frontend/src/interfaces/TimeEntry.ts
export interface TimeEntry {
  id: number;
  activity_id: number;
  start_time: string;
  end_time?: string;
  duration_seconds?: number;
}

export interface ActiveTimer {
  id: number;
  activity_id: number;
  activity_name: string;
  start_time: string;
  elapsed_seconds?: number; // Elapsed time from backend (excluding pauses)
  planned_duration?: number | null; // Duration in seconds, null = free-running
  paused_at?: string | null; // ISO timestamp when paused, null = not paused
  paused_duration?: number; // Cumulative seconds paused
  status?: 'running' | 'paused';
}
